CMakeLists.txt

```diff
@@ -38,13 +38,23 @@ if(WITH_MPI)
   endif()
 endif()
 
-if(WITH_OMP AND NOT TARGET OpenMP::OpenMP_Fortran)
-  find_package(OpenMP REQUIRED)
-  # Fix CMake bug on OpenMP settings for the NAG compiler
-  # See https://gitlab.kitware.com/cmake/cmake/-/issues/21280
-  if("${CMAKE_Fortran_COMPILER_ID}" STREQUAL "NAG" AND WITH_OMP)
-    set_property(TARGET OpenMP::OpenMP_Fortran PROPERTY INTERFACE_LINK_LIBRARIES "")
-    set_property(TARGET OpenMP::OpenMP_Fortran PROPERTY INTERFACE_LINK_OPTIONS "-openmp")
+if(WITH_OMP)
+  if(NOT TARGET OpenMP::OpenMP_Fortran)
+    find_package(OpenMP REQUIRED)
+    # Fix CMake bug on OpenMP settings for the NAG compiler
+    # See https://gitlab.kitware.com/cmake/cmake/-/issues/21280
+    if("${CMAKE_Fortran_COMPILER_ID}" STREQUAL "NAG" AND WITH_OMP)
+      set_property(TARGET OpenMP::OpenMP_Fortran PROPERTY INTERFACE_LINK_LIBRARIES "")
+      set_property(TARGET OpenMP::OpenMP_Fortran PROPERTY INTERFACE_LINK_OPTIONS "-openmp")
+    endif()
+  endif()
+else()
+  # Create empty targets, so that subprojects can depend on it without actual effect
+  if(NOT TARGET OpenMP::OpenMP_Fortran)
+    add_library(OpenMP::OpenMP_Fortran INTERFACE IMPORTED)
+  endif()
+  if(NOT TARGET OpenMP::OpenMP_C)
+    add_library(OpenMP::OpenMP_C INTERFACE IMPORTED)
```

Comment:
I don't think this is necessary with https://github.com/awvwgk/dftbplus/commit/3e0216ceb5df1a52ef26335ca97af9d9361c195b anymore.

Reply:
No, it isn't strictly necessary any more. Nevertheless, I think this would be a good practice on the DFTB+ side, to ease the integration of subprojects.